ledger/blockHeaderCache.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (C) 2019-2022 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package ledger
+
+import (
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/data/bookkeeping"
+	"github.com/algorand/go-deadlock"
+)
+
+const cacheSize = 512
```

Comment:
this isn't a great name for a global variable. But more important, I don't think that it should be a const anyway.
I believe that what you intended to have here is the CompactCertRounds * 2, and if so, we should make it depending on that value. Given that you need a constant value here, you could create a `MaxCompactCertRounds` in the config package and use it here.

Reply:
currently, we have a test that breaks if this number is not optimal for the current `CompactCertRounds`.

I'm not sure I like this value to be implicitly changed if the `CompactCertRounds` changes. 
From my perspective, we should have a test that breaks and someone explicitly changes it if needed.

Reply:
That's fine - in that case, I have two requests -;)
1. change the variable name to so that it would not potentially conflict. i.e. `blockHeadersSlidingWindowCacheSize` or similar.
2. add a comment on top of this constant, describing why this value was selected, and specifying the name of the unit test that validate the correctness of this value.